app/controllers/benefits/income/IncomeTaxBenefitsAmountController.scala

```diff
@@ -0,0 +1,122 @@
+  )
+
+  private def redirects(cya: EmploymentCYAModel, taxYear: Int, employmentId: String) = {
+    RedirectService.incomeTaxPaidByDirectorAmountRedirects(cya, taxYear, employmentId)
+  }
+
+  def show(taxYear: Int, employmentId: String): Action[AnyContent] = authAction.async { implicit user =>
+    inYearAction.notInYear(taxYear) {
+      employmentSessionService.getAndHandle(taxYear, employmentId) { (optCya, prior) =>
+
+        redirectBasedOnCurrentAnswers(taxYear, employmentId, optCya, EmploymentBenefitsType)(redirects(_, taxYear, employmentId)) { cya =>
+          val cyaAmount = cya.employment.employmentBenefits.flatMap(_.incomeTaxAndCostsModel.flatMap(_.incomeTaxPaidByDirector))
+
+          val form = fillFormFromPriorAndCYA(amountForm, prior, cyaAmount, employmentId)(
+            employment =>
+              employment.employmentBenefits.flatMap(_.benefits.flatMap(_.incomeTaxPaidByDirector))
+          )
+          Future.successful(Ok(incomeTaxBenefitsAmountView(taxYear, form, cyaAmount, employmentId)))
+        }
+      }
+    }
+
+  }
+
+  def submit(taxYear: Int, employmentId: String): Action[AnyContent] = authAction.async { implicit user =>
+    inYearAction.notInYear(taxYear) {
+
+      val redirectUrl = CheckYourBenefitsController.show(taxYear, employmentId).url
+
+      employmentSessionService.getSessionDataAndReturnResult(taxYear, employmentId)(redirectUrl) { cya =>
+
+        redirectBasedOnCurrentAnswers(taxYear, employmentId, Some(cya), EmploymentBenefitsType)(redirects(_, taxYear, employmentId)) { cya =>
+
+          amountForm.bindFromRequest().fold(
+            { formWithErrors =>
+              val cyaAmount = cya.employment.employmentBenefits.flatMap(_.incomeTaxAndCostsModel.flatMap(_.incomeTaxPaidByDirector))
+              Future.successful(BadRequest(incomeTaxBenefitsAmountView(taxYear, formWithErrors, cyaAmount, employmentId)))
+            }, {
+              amount =>
+
+                val cyaModel = cya.employment
+                val benefits = cyaModel.employmentBenefits
+                val incomeTaxModel = benefits.flatMap(_.incomeTaxAndCostsModel)
+
+                val updatedCyaModel = cyaModel.copy(
+                  employmentBenefits = benefits.map(_.copy(incomeTaxAndCostsModel =
+                    incomeTaxModel.map(_.copy(incomeTaxPaidByDirector = Some(amount)))))
+                )
+
+                employmentSessionService.createOrUpdateSessionData(employmentId, updatedCyaModel, taxYear,
+                  isPriorSubmission = cya.isPriorSubmission, hasPriorBenefits = cya.hasPriorBenefits)(errorHandler.internalServerError()) {
+
+                  if (cya.isPriorSubmission) {
+                    Redirect(CheckYourBenefitsController.show(taxYear, employmentId))
+                  } else {
+                    //TODO - redirect to the first reimbursed costs section question
```

Comment:
should this not go to the incurred costs question page? this page has been created, IncurredCostsBenefitsController, if you wanna remove the TODO and route to the page

Reply:
changed redirect